lib/Conch/Controller/DeviceReport.pm

```diff
@@ -19,54 +19,101 @@ with 'Conch::Role::MojoLog';
 
 =head2 process
 
-Processes the device report using the Legacy report code base
+Processes the device report, turning it into the various device_ tables as well
+as running validations
 
 =cut
 
 sub process ($c) {
-	my $device_report = $c->validate_input('DeviceReport');
-	return if not $device_report;
 	my $raw_report = $c->req->body;
+	my $schema = $c->schema;
 
-	my $maybe_hw;
+	my $unserialized_report = $c->validate_input('DeviceReport');
+	if(not $unserialized_report) {
+		return; # JSON validator will throw the proper error response
```

Comment:
technically nothing is "thrown" - the response has just already been set in $c->tx->res.

Reply:
Addressed